src/object.d

```diff
@@ -2054,7 +2054,12 @@ auto byKey(T : V[K], K, V)(T aa) pure nothrow @nogc @safe
         @property Result save() { return this; }
     }
 
-    return Result(_aaRange(() @trusted { return cast(void*)aa; } ()));
+    // ensure we are dealing with a genuine AA.
+    static if(is(V[K] == T))
+        alias realAA = aa;
+    else
+        const(V[K]) realAA = aa;
+    return Result(_aaRange(() @trusted { return cast(void*)realAA; } ()));
```

Comment:
How about adding a template version of `_aaRange`, so you can encapsulate the logic there?